### PR TITLE
Add Hazelcast Helm Chart

### PR DESCRIPTION
**Pull request by leszko**
**What this PR does / why we need it**:
Add Helm Chart for Hazelcast IMDG (https://hazelcast.org/)

**Which issue this PR fixes**: 
N/A

**Special notes for your reviewer**:
This is an initial PR for Hazelcast, which will be maintained by the Hazelcast Integration Team (exact names are mentioned in Charts.yaml and OWNERS).

### CODE REVIEW

File: stable/hazelcast/templates/rbac.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{{- if .Values.rbac.create -}}
```
Comment (unguiculus) [inline stable/hazelcast/templates/rbac.yaml]: Please follow our best practices for RBAC:
https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md

File: stable/hazelcast/values.yaml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  timeoutSeconds: 1
+  # successThreshold is the minimum consecutive successes for the probe to be considered successful after having failed
+  successThreshold: 1
+  # failureThreshold is the minimum consecutive failures for the probe to be considered failed after having succeeded
+  failureThreshold: 3
+
+# Configure resource requests and limits
+# ref: http://kubernetes.io/docs/user-guide/compute-resources/
+#
+resources:
```
Comment (unguiculus) [inline stable/hazelcast/values.yaml]: Don't specify defaults for resources. Comment them out.

File: stable/hazelcast/.helmignore [hunk 3]
```diff
@@ -0,0 +1 @@
+.git
```
Comment (unguiculus) [inline stable/hazelcast/.helmignore]: Please use the `.helmignore` file you get from `helm create`.

File: stable/hazelcast/templates/statefulset.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
+apiVersion: apps/v1beta1
```
Comment (unguiculus) [inline stable/hazelcast/templates/statefulset.yaml]: Do you need to support K8s < 1.9? Otherwise, please use `apps/v1`.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/hazelcast/templates/rbac.yaml (hunk 1 above)
Comment (leszko) [reply stable/hazelcast/templates/rbac.yaml]: Done.

File: stable/hazelcast/values.yaml (hunk 2 above)
Comment (leszko) [reply stable/hazelcast/values.yaml]: Done

File: stable/hazelcast/.helmignore (hunk 3 above)
Comment (leszko) [reply stable/hazelcast/.helmignore]: Done

File: stable/hazelcast/templates/statefulset.yaml (hunk 4 above)
Comment (leszko) [reply stable/hazelcast/templates/statefulset.yaml]: Done
We don't need K8 < 1.9.

**Comment by leszko** [comment]
@unguiculus Thanks for the comments. I've addressed all of them. PTAL.

**Comment by unguiculus** [comment]
One more thing. Please use a consistent list indentation style. Either indent zero or two spaces throughout the chart.

**Comment by leszko** [comment]
List indentation fixed.

**Comment by leszko** [comment]
@unguiculus , should I fix anything more in this PR?

**Comment by gregrluck** [comment]
Hey guys, this is our second attempt to get our helm chart PR done. You stopped responding completely on https://github.com/helm/charts/pull/6248 so we did another one. Would appreciate it you could please take a look. We believe this is ready to merge as we have fixed all review points.

**Pull request merged**